src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -166,7 +167,7 @@ public IUrlHelper Url
             {
                 if (_url == null)
                 {
-                    _url = Resolver?.GetRequiredService<IUrlHelper>();
+                    _url = Resolver?.GetRequiredService<IUrlHelperFactory>()?.Create(ControllerContext);
```

Comment:
Huh? Combo of `GetRequiredService` and the null-propagating operator implies a whole new physics.